test/persistence-hooks.suite.js

```diff
@@ -3178,15 +3181,21 @@ module.exports = function(dataSource, should, connectorCapabilities) {
           { id: 'new-id', name: 'a name' },
           function(err, instance) {
             if (err) return done(err);
-            ctxRecorder.records.should.eql(aCtxForModel(TestModel, {
+            var expectedContext = aCtxForModel(TestModel, {
               where: { id: 'new-id' },
               data: { id: 'new-id', name: 'a name' },
               currentInstance: {
                 id: 'new-id',
                 name: 'a name',
                 extra: undefined,
               },
-            }));
+            });
+
+            if (dataSource.connector.upsertWithWhere) {
+              expectedContext.where = {
+                id: 'new-id',
```

Comment:
I find this suspicious. AFAICT, `expectedContext.where` is already initalised with `{ id: 'new-id' }`.